scripts/filter_dataset.py

```diff
@@ -0,0 +1,179 @@
+            prompt.append({"role": "system", "content": training_args.system_prompt})
+
+        if prompt_column not in example:
+            raise ValueError(f"Dataset Question Field Error: {prompt_column} is not supported.")
+
+        prompt.append({"role": "user", "content": example[prompt_column]})
+        return {"prompt": prompt}
+
+    dataset = dataset.map(make_conversation)
+    tokenizer = get_tokenizer(model_args, training_args)
+    for split in dataset:
+        if "messages" in dataset[split].column_names:
+            dataset[split] = dataset[split].remove_columns("messages")
+    
+    dataset = dataset.map(apply_chat_template, fn_kwargs={"tokenizer": tokenizer})
+    llm = LLM(
+        model=model_args.model_name_or_path,
+        revision=model_args.model_revision,
+        trust_remote_code=model_args.trust_remote_code,
+    )
+
+    sampling_params=SamplingParams(
+        temperature=training_args.temperature,
+        n=training_args.num_generations,
+        max_tokens=training_args.max_completion_length,
+    )
+    
+    def batch_score(examples):
+        prompts = examples["prompt"]
+        
+        outputs = llm.generate(
+            prompts,
+            sampling_params=sampling_params,
+            use_tqdm=False,
+        )
+        repeated_prompts = []
+        reward_completions = []
+        grouped_completions = []
+        for output in outputs:
+            prompt = output.prompt
+            group = []
+            for completion in output.outputs:
+                text = completion.text
+                group.append(text)
+                message = [{"role": "assistant", "content": text}]
+                repeated_prompts.append(prompt)
+                reward_completions.append(message)
+            grouped_completions.append(group)
+        
+        def repeat_each_element_k_times(list_to_repeat: list, k: int) -> list:
+            return [element for item in list_to_repeat for element in [item] * k]
+        
+        rewards_per_func = torch.zeros(len(repeated_prompts), len(reward_funcs))
+        for i, reward_func in enumerate(reward_funcs):
+            keys = [key for key in examples.data.keys() if key not in ["prompt", "completion"]]
+            reward_kwargs = {key: repeat_each_element_k_times(examples[key], training_args.num_generations) for key in keys}
+            output_reward_func = reward_func(prompts=repeated_prompts, completions=reward_completions, **reward_kwargs)
+            # Convert None values to NaN
+            output_reward_func = [reward if reward is not None else torch.nan for reward in output_reward_func]
+
+            rewards_per_func[:, i] = torch.tensor(output_reward_func, dtype=torch.float32)
+            
+        reshaped_rewards = rewards_per_func.view(-1, training_args.num_generations)
+        
+        examples["filter_generations"] = grouped_completions
+        examples["filter_rewards"] = reshaped_rewards.tolist()
+            
```

Comment:
```suggestion
            example["pass_rate_generator"] = model_args.model_name_or_path
```

Reply:
I was thinking to change the dataset name to include the model and revision instead?